pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -215,6 +232,20 @@ func (d *DeploymentController) Reconcile(req types.NamespacedName) error {
 		if !knownControllers.Contains(string(gc.Spec.ControllerName)) {
 			return nil
 		}
+		// find the tag or revision indicated by the object
+		selectedTag, ok := gw.Labels[label.IoIstioRev.Name]
+		if !ok {
+			ns := d.namespaces.Get(gw.Namespace, "")
+			if ns == nil {
+				return nil
+			}
+			selectedTag = ns.Labels[label.IoIstioRev.Name]
+		}
+		myTags := d.tagWatcher.GetMyTags()
```

Comment:
do we need to consider the label on the Gateway itself?

Reply:
I think we're doing that at L238